### DISCUSSION TITLE
How to include a static lightmap in a material definition

### DISCUSSION THREAD

**Discussion by rawnsley** [google/filament#4650] [Ideas]
In three.js some materials have a [lightmap property](https://threejs.org/docs/#api/en/materials/MeshStandardMaterial.lightMap) that is used as a multiplier for indirect light sources.

I'm using some GLTF extensions to bring in HDR lightmap map textures that I have baked in Blender using Cycles, but I'm struggling to work out how I should include them in my material definitions in Filament.

In unlit mode I can just multiply the materials `baseColor`, but in lit mode this doesn't affect the specular light components from the IBL as I would like it to.

**Comment by romainguy** [comment]
Copied from the bug, to continue the discussion:

We don't support lightmaps currently because we haven't decided how we'd like our lightmaps to work (the solution you describe does not provide any directionality, we'd prefer to use something based on spherical harmonics or gaussians). Proper lightmaps would also require a baking tool.

How do you generate your lightmaps btw and what form do they take? Are they just a single channel luminance map or a colored map?

**Reply by romainguy** [reply]
One hack you could try is `postLightingColor`/`postLightingBlending`: https://google.github.io/filament/Materials.html#materialmodels/litmodel/post-lightingcolor and https://google.github.io/filament/Materials.html#materialdefinitions/materialblock/blendingandtransparency:postlightingblending. The `multiply` mode might do what you want?

**Reply by rawnsley** [reply]
This works for me - thank you. I guess it would be applied erroneously to emissive lights, but I can live with that as we wouldn't typically add a lightmap over the top of an emissive material anyway.
